packages/db/src/collection.ts

```diff
@@ -335,6 +335,7 @@ export class CollectionImpl<
   private gcTimeoutId: ReturnType<typeof setTimeout> | null = null
   private preloadPromise: Promise<void> | null = null
   private syncCleanupFn: (() => void) | null = null
+  private cleanupInFlight: Promise<void> | null = null
```

Comment:
I don't like that this can be a promise or null. I'd rather have it always as a promise. So the start value here would be `Promise.resolve<void>()`. This would simplify the handling of `cleanupInFlight` as we always just await it. This also means we no longer have to set this to `null` in the finally blocks on L720 and L728.